controllers/vspherecluster_controller.go

```diff
@@ -330,7 +359,60 @@ func cloudProviderConfigurationAvailable(ctx *context.ClusterContext) bool {
 	return !reflect.DeepEqual(ctx.VSphereCluster.Spec.CloudProviderConfiguration, infrav1.CPIConfig{})
 }
 
+func (r clusterReconciler) reconcileIdentitySecret(ctx *context.ClusterContext) error {
+	vsphereCluster := ctx.VSphereCluster
```

Comment:
can we add a test to cover secret reconciliation ? (happy path + maybe a case secret where the secret is already owned)